### ISSUE TITLE
sct_flatten_sagittal crashes

### ISSUE THREAD

**Issue by stephaniealley** [neuropoly/spinalcordtoolbox#1726]
### Description

The function crashes and produces the error below.

### Steps to Reproduce

Data: sct_testing/issues/20180513_issue1726

Run:
```
sct_flatten_sagittal -i t2.nii.gz -s t2_centerline.nii.gz
```

Error: 
```
Running /Users/Storme/sct/scripts/sct_flatten_sagittal.py -i t2.nii.gz -s t2_centerline.nii.gz

Check input arguments...
  Input volume ...................... t2.nii.gz
  Centerline ........................ t2_centerline.nii.gz


Get dimensions of input centerline...
.. matrix size: 52 x 60 x 55
.. voxel size:  1.0mm x 1.0mm x 1.0mm

Open centerline volume...
Test: # of control points = 5
Error on approximation = 0.06 mm
Test: # of control points = 6
WARNING: NURBS instability -> wrong reconstruction
Test: # of control points = 7
WARNING: NURBS instability -> wrong reconstruction
Test: # of control points = 8
Error on approximation = 0.05 mm
Test: # of control points = 9
Error on approximation = 0.04 mm
Test: # of control points = 10
Error on approximation = 0.02 mm
Test: # of control points = 11
Error on approximation = 0.03 mm
Test: # of control points = 12
Error on approximation = 0.02 mm
Number of control points of the optimal NURBS = 12

Split input volume...
Traceback (most recent call last):
  File "/Users/Storme/sct/scripts/sct_flatten_sagittal.py", line 325, in <module>
    main(fname_anat, fname_centerline, degree_poly, centerline_fitting, interp, remove_temp_files, verbose)
  File "/Users/Storme/sct/scripts/sct_flatten_sagittal.py", line 138, in main
    displacement_max_z_index = x_centerline_fit[z_init - min_z_index] - x_centerline_fit[max_z_index - min_z_index]
IndexError: index 54 is out of bounds for axis 0 with size 54

```

### State of spinalcordtoolbox
```
master/75a7408fd699da5966f5ef18d7188403145fa4fb
```

**Comment by fperdigon** [comment]
@stephaniealley this is related to #821. I am working on this

**Comment by stephaniealley** [comment]
@fperdigon thank you for letting me know.

**Comment by fperdigon** [comment]
@stephaniealley @jcohenadad  I think that this issue can be closed

**Comment by jcohenadad** [comment]
fixed in #1729
